src/v2/cookbook/form-validation.md

```diff
@@ -0,0 +1,322 @@
+    <b>Please correct the following error(s):</b>
+    <ul>
+      <li v-for="error in errors">{{ error }}</li>
+    </ul>
+  </p>
+  
+  <p>
+    <label for="name">Name<label>
+    <input type="text" name="name" id="name" v-model="name">
+  </p>
+
+  <p>
+    <label for="email">Email<label>
+    <input type="email" name="email" id="email" v-model="email">
+  </p>
+
+  <p>
+    <label for="movie">Favorite Movie<label>
+    <select name="movie" id="movie" v-model="movie">
+      <option>Star Wars</option>
+      <option>Vanilla Sky</option>
+      <option>Atomic Blonde</option>
+    </select>
+  </p>
+
+  <p>
+    <input type="submit" value="Submit">  
+  </p>
+
+</form>
+```
+
+While the change here is small, note the `novalidate="true"` on top. This is important because the browser will attempt to validate the email address in the field when type="email". Frankly it may make more sense to trust the browser in this case, but as we wanted an example with custom validation, we're disabling it. Here's the updated JavaScript.
+
+``` js
+const app = new Vue({
+  el:'#app',
+  data:{
+    errors:[],
+    name:null,
+    email:null,
+    movie:null
+  },
+  methods:{
+    checkForm:function(e) {
+      this.errors = [];
+      if(!this.name) this.errors.push("Name required.");
+      if(!this.email) {
+        this.errors.push("Email required.");
+      } else if(!this.validEmail(this.email)) {
+        this.errors.push("Valid email required.");        
+      }
+      if(!this.errors.length) return true;
+      e.preventDefault();
+    },
+    validEmail:function(email) {
+      var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(email);
+    }
+  }
+})
+```
+
+As you can see, we've added `validEmail` as a new method and it is simply called from `checkForm`. You can play with this example here:
+
+<p data-height="265" data-theme-id="0" data-slug-hash="vWqNXZ" data-default-tab="html,result" data-user="cfjedimaster" data-embed-version="2" data-pen-title="form validation 2" class="codepen">See the Pen <a href="https://codepen.io/cfjedimaster/pen/vWqNXZ/">form validation 2</a> by Raymond Camden (<a href="https://codepen.io/cfjedimaster">@cfjedimaster</a>) on <a href="https://codepen.io">CodePen</a>.</p>
+<script async src="https://production-assets.codepen.io/assets/embed/ei.js"></script>
+
+## Another Example of Custom Validation
+
+For the third example, we've built something you've probably seen in survey apps. The user is asked to spend a "budget" for a set of features for a new Star Destroyer model. The total must equal 100. First, the HTML.
+
+``` html
+<form id="app" @submit="checkForm" action="/something" method="post" novalidate="true">
+  
+  <p v-if="errors.length">
+    <b>Please correct the following error(s):</b>
+    <ul>
+      <li v-for="error in errors">{{ error }}</li>
+    </ul>
+  </p>
+
+  <p>

```

Comment:
I quite like this example, but I think the text here could make it clearer that you have to select exactly 100. Maybe something like "Given a budget of 100 dollars, indicate how much you would spend on the following features for the next generation Star Destroyer by making the Current Total of features add up to 100:"

Reply:
I went with: " Your total must sum up to 100.".